### PR TITLE
Watchlistbackend

### PR DESCRIPTION
**Pull request by martinupset**
I like working fellow yuan, I like his structure n api, he is a good coder! I

### CODE REVIEW

**Comment by adamgyulavari** [comment]
@martinupset fix the conflicts and close the previous PR-s if they're not needed anymore

File: backend-opal/src/database/migrations/1574590998449_create_table_users.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module.exports = {
```
Comment (adamgyulavari) [inline backend-opal/src/database/migrations/1574590998449_create_table_users.js]: @martinupset let's call this migration create table watchlists

File: backend-opal/src/database/migrations/1574590998449_create_table_users.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  up: `CREATE TABLE WatchList (
```
Comment (adamgyulavari) [inline backend-opal/src/database/migrations/1574590998449_create_table_users.js]: @martinupset 
```suggestion
  up: `CREATE TABLE watch_lists (
```

File: backend-opal/src/server/db/driver.js [hunk 3]
```diff
@@ -52,6 +52,12 @@ function getStockByUserId(id) {
   return executeQuery(db, query);
 }
 
+function getSymbols(userId) {
```
Comment (adamgyulavari) [inline backend-opal/src/server/db/driver.js]: @martinupset getWatchListsByUserId

File: backend-opal/src/server/db/driver.js [hunk 4]
```diff
@@ -52,6 +52,12 @@ function getStockByUserId(id) {
...
+  const sql = `select symbols from watchlist where user_id = ${db.escape(userId)}`;
```
Comment (adamgyulavari) [inline backend-opal/src/server/db/driver.js]: @martinupset 
```suggestion
  const sql = `select * from watchlist where user_id = ${db.escape(userId)}`;
```

File: backend-opal/src/server/watchListService.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+const driver = require('./db/driver');
+
+const getSymbols = async userID => {
```
Comment (adamgyulavari) [inline backend-opal/src/server/watchListService.js]: @martinupset getWatchLists

File: backend-opal/src/server/watchLists.js [hunk 6]
```diff
@@ -1,23 +1,16 @@
...
+const WatchListsData = require('./watchListService');
 
 function checkRequestBody(body) {
   const bodyKey = ['userId', 'name'];
   const result = bodyKey.filter(key => !(key in body));
   return result;
 }
 
-router.get('/', function(req, res) {
+router.get('/:UserId', async (req, res) => {
```
Comment (adamgyulavari) [inline backend-opal/src/server/watchLists.js]: @martinupset don't need the userId in the route

File: backend-opal/src/server/watchLists.js [hunk 7]
```diff
@@ -32,7 +25,7 @@ router.get('/', function(req, res) {
     res.status(403).send(response);
   } else {
     const response = {
-      watchlists: data
+      watchlists: await WatchListsData.getSymbols(UserId)
```
Comment (adamgyulavari) [inline backend-opal/src/server/watchLists.js]: @martinupset just use 
```suggestion
      watchlists: await WatchListsData.getSymbols(1)
```

File: backend-opal/src/server/watchLists.js [hunk 8]
```diff
@@ -1,23 +1,16 @@
...
 function checkRequestBody(body) {
   const bodyKey = ['userId', 'name'];
   const result = bodyKey.filter(key => !(key in body));
   return result;
 }
 
-router.get('/', function(req, res) {
+router.get('/:UserId', async (req, res) => {
   const { authorization } = req.headers;
+  const { UserId } = req.params;
```
Comment (adamgyulavari) [inline backend-opal/src/server/watchLists.js]: @martinupset this userId should come from the header

File: frontend-opal/App.js [hunk 9]
```diff
@@ -1,7 +1,14 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
```
Comment (adamgyulavari) [inline frontend-opal/App.js]: @martinupset this file shouldnt be changed

File: backend-opal/src/server/db/driver.js [hunk 10]
```diff
@@ -52,6 +45,12 @@ function getStockByUserId(id) {
   return executeQuery(db, query);
 }
 
+function getSymbols(getWatchListsByUserId) {
```
Comment (adamgyulavari) [inline backend-opal/src/server/db/driver.js]:
```suggestion
function getWatchListsByUserId(userId) {
```

File: backend-opal/src/server/watchListService.js [hunk 11]
```diff
@@ -0,0 +1,9 @@
+const driver = require('./db/driver');
+
+const getSymbols = async getWatchlists => {
```
Comment (adamgyulavari) [inline backend-opal/src/server/watchListService.js]:
```suggestion
const getWatchlists = async userId => {
```

File: backend-opal/src/server/watchLists.test.js [hunk 12]
```diff
@@ -1,9 +1,10 @@
+jest.mock('./watchListService');
 const request = require('supertest');
 const server = require('../../server');
 
 describe('Watchlist Endpoint', () => {
   it('no authorization, should get 401', async () => {
-    const res = await request(server).get('/watchlists');
+    const res = await request(server).get('/watchlists/1');
```
Comment (adamgyulavari) [inline backend-opal/src/server/watchLists.test.js]: @martinupset 
```suggestion
    const res = await request(server).get('/watchlists');
```

File: frontend-opal/App.js [hunk 13]
```diff
@@ -10,9 +9,11 @@ export default function App() {
...
   const store = createStore(reducers, composeEnhancers(applyMiddleware(thunk)));
 
-  return (
-    <Provider store={store}>
-      <Investment />
-    </Provider>
+  
+
+  return ( 
+        <Provider store={store}>
```
Comment (adamgyulavari) [inline frontend-opal/App.js]: @martinupset this shouldnt be here

**Comment by adamgyulavari** [comment]
@martinupset fix conflicts (locally)

File: backend-opal/src/database/migrations/1574590998449_create_table_users.js (hunk 1 above)
Comment (martinupset) [reply backend-opal/src/database/migrations/1574590998449_create_table_users.js]: ok

File: backend-opal/src/database/migrations/1574590998449_create_table_users.js (hunk 2 above)
Comment (martinupset) [reply backend-opal/src/database/migrations/1574590998449_create_table_users.js]: ok

File: backend-opal/src/server/db/driver.js (hunk 3 above)
Comment (martinupset) [reply backend-opal/src/server/db/driver.js]: ok

File: backend-opal/src/server/db/driver.js (hunk 4 above)
Comment (martinupset) [reply backend-opal/src/server/db/driver.js]: ok

File: backend-opal/src/server/watchListService.js (hunk 5 above)
Comment (martinupset) [reply backend-opal/src/server/watchListService.js]: ok

File: backend-opal/src/server/watchLists.js (hunk 6 above)
Comment (martinupset) [reply backend-opal/src/server/watchLists.js]: ok

File: backend-opal/src/server/watchLists.js (hunk 7 above)
Comment (martinupset) [reply backend-opal/src/server/watchLists.js]: ok

File: backend-opal/src/server/watchLists.js (hunk 8 above)
Comment (martinupset) [reply backend-opal/src/server/watchLists.js]: ok

File: frontend-opal/App.js (hunk 9 above)
Comment (martinupset) [reply frontend-opal/App.js]: ok

File: backend-opal/src/server/db/driver.js (hunk 10 above)
Comment (martinupset) [reply backend-opal/src/server/db/driver.js]: ok

File: backend-opal/src/server/watchListService.js (hunk 11 above)
Comment (martinupset) [reply backend-opal/src/server/watchListService.js]: ok

File: backend-opal/src/server/watchLists.test.js (hunk 12 above)
Comment (martinupset) [reply backend-opal/src/server/watchLists.test.js]: ok

File: frontend-opal/App.js (hunk 13 above)
Comment (martinupset) [reply frontend-opal/App.js]: ok

**Comment by martinupset** [comment]
done

File: backend-opal/src/server/watchLists.js [hunk 14]
```diff
@@ -1,28 +1,20 @@
...
-  {
-    id: 1,
-    userId: 1,
-    name: 'My Watchlist 1',
-    symbols: ['MSFT', 'AMZN', 'AAPL', 'FB']
-  }
-];
+router.get('/', async (req, res) => {
+  const response = {
+    portfolio: await WatchListsData.getWatchLists(1)
```
Comment (adamgyulavari) [inline backend-opal/src/server/watchLists.js]: @martinupset `watchlists:` :D

**Pull request merged**